core/unit_test/incremental/Test01_execspace.hpp

```diff
@@ -104,7 +104,7 @@ TEST(TEST_CATEGORY, IncrTest_01_execspace_typedef) {
 }
 
 TEST(TEST_CATEGORY, IncrTest_01_execspace) {
-  ASSERT_TRUE(Kokkos::is_execution_space<TEST_EXECSPACE>::value);
+  ASSERT_FALSE(!Kokkos::is_execution_space<TEST_EXECSPACE>::value);
```

Comment:
What was wrong with `ASSERT_TRUE`?

Reply:
Using `ASSERT_FALSE` is a workaround. After the update, this line would cause:
```
Kokkos  kokkos  build  1  /bin/clang++ -Xclang -v -Xlinker -v -Werror -g -DKOKKOS_DEPENDENCE -fopenmp=libomp -Wall -Wunused-parameter -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/UnitTestMainInit.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test01_execspace_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test02_atomic_host_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test03a_MemorySpace_malloc_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test03b_MemorySpace_free_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test04_ParallelFor_RangePolicy_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test05_ParallelReduce_RangePolicy_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test06_ParallelFor_MDRangePolicy_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test08_deep_copy_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test10_HierarchicalBasics_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test11a_ParallelFor_TeamThreadRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test11b_ParallelFor_TeamVectorRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test11c_ParallelFor_ThreadVectorRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test12a_ThreadScratch_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test12b_TeamScratch_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test13a_ParallelRed_TeamThreadRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test13b_ParallelRed_TeamVectorRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test13c_ParallelRed_ThreadVectorRange_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test14_MDRangeReduce_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test16_ParallelScan_OPENMP.cpp.o core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test17_CompleteAtomic_OPENMP.cpp.o -o core/unit_test/KokkosCore_IncrementalTest_OPENMP  core/unit_test/libkokkos_gtest.a  containers/src/libkokkoscontainers.a  core/src/libkokkoscore.a  /usr/lib/x86_64-linux-gnu/libhwloc.so  /usr/lib/x86_64-linux-gnu/libdl.so
GNU ld (GNU Binutils for Ubuntu) 2.36.1
/bin/ld: /bin/ld: DWARF error: could not find variable specification at offset 56
/bin/ld: DWARF error: could not find variable specification at offset 122
core/unit_test/CMakeFiles/KokkosCore_IncrementalTest_OPENMP.dir/generated/Test01_execspace_OPENMP.cpp.o: in function `Test::OPENMP_IncrTest_01_execspace_Test::TestBody()':
/home/cz4rs/Kokkos/kokkos/build/../core/unit_test/incremental/Test01_execspace.hpp:107: undefined reference to `Kokkos::is_execution_space<Kokkos::OpenMP>::value'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

As far as I can see there is nothing wrong with `Kokkos::is_execution_space` and the problem lies in GoogleTest. I didn't find the root cause, but I have discovered that `ASSERT_FALSE` does not have this problem.